src/Xamarin.CodeAnalysis/RequiresSuperAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,92 @@
+			var diagnostic = context.Diagnostics.First ();
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+			// Find the method declaration identified by the diagnostic.
+			var methodDeclaration =
+			  root.FindToken (diagnosticSpan.Start).Parent.AncestorsAndSelf ()
+			  .OfType<MethodDeclarationSyntax> ().First ();
+
+			// Register a code action that will invoke the fix.
+			context.RegisterCodeFix (CodeAction.Create (Title, c => FixSuperAsync (context.Document, methodDeclaration, c), equivalenceKey: Title), diagnostic);
+		}
+
+		private async Task<Document> FixSuperAsync (Document document,
+		  MethodDeclarationSyntax methodDeclaration,
+		  CancellationToken cancellationToken)
+		{
+			//Find the node to fix.
+			var semanticModel = await document.GetSemanticModelAsync (cancellationToken);
+			var memberSymbol = semanticModel.GetDeclaredSymbol (methodDeclaration) as IMethodSymbol;
+
+			//Generate the new base.[methodName]([arguments]) block. If the method needs to return the base call,
+			//it generates return base.[methodName]([arguments])
+			// TODO: this might be hacky - (.Add(SyntaxFactory.Tab) adds an extra tab so that it's a tab (or SyntaxFactory.Tab) further than the leading
+			//      trivia of the method body block, but we should consider investigating if this behaves as expected when the IDE is set to spaces vs
+			//      tabs / different numbers of spaces per tab
+			var returnString = memberSymbol.ReturnsVoid ? "" : "return ";
+			var parametersString = string.Join (",", memberSymbol.Parameters.Select ((p) => p.Name).ToArray ()); //creates a string of the method's passed-in arguments separated by commas
+			var newLiteral = SyntaxFactory.ParseStatement (returnString + "base." + memberSymbol.Name + "(" + parametersString + ");")
+			  .WithLeadingTrivia (methodDeclaration.Body.GetLeadingTrivia ().Add (SyntaxFactory.Tab))
+			  .WithTrailingTrivia (methodDeclaration.Body.GetTrailingTrivia ())
+			  .WithAdditionalAnnotations (Formatter.Annotation);
+			BlockSyntax blockLiteralWithBraces = SyntaxFactory.Block (newLiteral);
+
+			//Get rid of the extra { } braces created by the block creation.
+			//TODO: this is hacky ({ } braces are automatically added when you make a new BlockSyntax, and this gets of them but still keeps them in the syntax
+			//      tree. We should consider investigating how to do use something else rather than blocks, or how to create a block without enclosing braces.
+			BlockSyntax blockLiteral = blockLiteralWithBraces.WithOpenBraceToken (SyntaxFactory.MissingToken (SyntaxKind.OpenBraceToken))
+			.WithCloseBraceToken (SyntaxFactory.MissingToken (SyntaxKind.CloseBraceToken));
+
+			//Add new block to the pre-existing block.
+			BlockSyntax methodDeclarationBlock = methodDeclaration.Body.AddStatements (blockLiteral);
+
+			//Swap new node into syntax tree.
+			var root = await document.GetSyntaxRootAsync ();
+			var newRoot = root.ReplaceNode (methodDeclaration.Body, methodDeclarationBlock);
```

Comment:
Q: does this replace the whole method body with the new block ?

e.g. does

```csharp
override void Foo ()
{
   Console.WriteLine ("foo");
}
```

become
```csharp
override void Foo ()
{
   base.Foo ();
}
```

or 
```csharp
override void Foo ()
{
   Console.WriteLine ("foo");
   base.Foo ();
}
```

The corresponding ObjC macro _requires super_ means that a call is required - but something else might happen before, or after, that call.

Reply:
@madelinez820 has a fairly long list of test cases that she's tried out :)

There is a test framework for Roslyn analyzers, some examples are in this repo: https://github.com/xamarin/CodeAnalysis/tree/master/src/Xamarin.CodeAnalysis.Tests

If there's time before the end of the externship, tests may get added in a separate PR 😉

Reply:
I can set up the test framework, either before or after this PR.

Reply:
The tests seem to rely on a `Android.Runtime.ResourceDesignerAttribute` attribute. Which NuGet package is it in?

Reply:
@sharwell - @madelinez820's analyzer is for the Xamarin MaciOS SDK. This PR is the first MaciOS analyzer in this repo (other analyzers in CodeAnalysis appear to be for Android). I don't know how that affects setup for test frameworks, but I would guess that there would not be a dependency on Android resources.

@kzu - can you provide more info on the existing tests and what NuGet packages they use?

Reply:
@whitneyschmidt Thanks! There are two existing analyzer tests, which I was trying to convert as an example of how to use the new test library to write analyzer tests. Those use Android. If I don't convert them, I could simply add a new test for this analyzer to form the example as long as I knew which TFM and package(s) it needs.

Reply:
@sharwell you need a VS installation with the Xamarin workload in place. See https://github.com/xamarin/CodeAnalysis/blob/master/src/Xamarin.CodeAnalysis.Tests/Xamarin.CodeAnalysis.Tests.csproj#L26. 

There is no nuget package with the XA SDK (yet? in the .NET5 timeframe almost for sure this would be easier to acquire, I suppose).

Reply:
@spouliot - to answer your original question (sorry I missed it previously), it doesn't replace the existing stuff in the body (methodDeclarationBlock includes what was previously there).